lib/rules/require-prop-types.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @fileoverview Prop definitions should be detailed
+ * @author Armano
+ */
+'use strict'
+
+const utils = require('../utils')
+
+function create (context) {
+  // ----------------------------------------------------------------------
+  // Helpers
+  // ----------------------------------------------------------------------
+
+  function objectHasType (node) {
+    const typeProperty = node.properties
+      .filter(p =>
+        utils.getStaticPropertyName(p.key) === 'type' &&
+        (
+          p.value.type !== 'ArrayExpression' ||
+          p.value.elements.length > 0
+        )
+      )[0]
+    return Boolean(typeProperty)
+  }
+
+  function checkProperties (items) {
+    items.map(cp => {
```

Comment:
I'm wondering that it uses `Array#map` to iterate the array. I like `for-of` :smiley: